clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2473,9 +2475,116 @@ class SyclKernelIntHeaderCreator : public SyclKernelFieldHandler {
 
 } // namespace
 
+class SYCLKernelNameTypeVisitor
+    : public TypeVisitor<SYCLKernelNameTypeVisitor>,
+      public ConstTemplateArgumentVisitor<SYCLKernelNameTypeVisitor> {
+  Sema &S;
+  SourceLocation Loc;
```

Comment:
Lets make name of this member a bit more informative
```suggestion
  SourceLocation KernelCallerLoc;
```